lib/ansible/modules/cloud/amazon/s3.py

```diff
@@ -713,22 +713,18 @@ def main():
 def get_s3_connection(aws_connect_kwargs, location, rgw, s3_url):
     if s3_url and rgw:
         rgw = urlparse(s3_url)
-        s3 = boto.connect_s3(
-            is_secure=rgw.scheme == 'https',
-            host=rgw.hostname,
-            port=rgw.port,
-            calling_format=OrdinaryCallingFormat(),
-            **aws_connect_kwargs
-        )
+        aws_connect_kwargs['is_secure'] = rgw.scheme == 'https'
```

Comment:
There are a couple of cases when we need to use ordinary calling format- if the bucket name has a . in it and if fakes3 or rgw are being used. But we don't need to call it in Line531 if we'll be using it later for fakes3 or rgw, because then we'd be passing the connection calling_format twice, which breaks it. Why is that line at fundamental odds with get_s3_connection? It appears it was overlooked that rgw or fakes3 could have a . in the bucket name.